Modal to force username verification

SabreCat opened the pull request:
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

**NOT FOR PRODUCTION AT THIS TIME** - This is meant to go live when we are ready to require usernames for all users, in preparation for full feature rollout.

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)

Fixes #10697 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

* Adds a modal that appears automatically if the user has not yet verified their username. The modal gives the user the opportunity to change their display name and/or username and cannot be dismissed except by providing valid input for both fields.
* Adds an API route `verify-display-name`, based on `verify-username`, so we can check display names for slurs and length inline.
* Cleans out a few unused name-related strings.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

SabreCat committed:
feat(usernames): modal to force verification

SabreCat committed:
fix(test): string is miscapitalized :(

SabreCat committed:
fix(usernames): remove more ratzen fratzen dupe strings

SabreCat committed:
fix(test): copypasta and string trouble

SabreCat committed:
fix(username): Add @ prepend

SabreCat commented:
If the "verify display name" implementation looks OK, I can implement it over on the Settings page as well.

SabreCat committed:
fix(usernames): add display name verification to site settings
also corrrect some validation logic in force-verify modal

SabreCat committed:
fix(settings): don't center align display name issues

SabreCat committed:
fix(usernames): various
Better modal positioning
Correct text colors for input field
Don't show "already taken" if username has other errors

SabreCat committed:
feat(usernames): vet pet announcement

SabreCat committed:
fix(usernames): let verify modal grow to content

paglias commented on website/client/components/settings/verifyUsername.vue:
```diff
@@ -0,0 +1,270 @@
+      ...mapState({
+        user: 'user.data',
+      }),
+      displayNameInvalid () {
```
I think usernames and display names with a lenght of 1 should be valid, here they're valid only with at least 2 chars

paglias commented:
I'm having an issue where on this branch the RYA modal never opens

SabreCat committed:
fix(usernames): various
z-index modals above Resting banner
force reload after verify username
add missing e-mail validation on frontpage
let Yesterdaily modal float behind username modal

SabreCat commented:
I'm going to combine this PR with the other one, as they're all going live at the same time and I otherwise have to keep merging them together to test, hah
